src/EditorFeatures/Core/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -237,6 +244,27 @@ public IEnumerable<char> PotentialCommitCharacters
                 return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
             }
 
+            // Specifically for snippets, we use reflection to try and invoke the LanguageServerSnippetExpander's
+            // TryExpand method and determine if it succeeded or not.
+            if (SnippetCompletionItem.IsSnippet(roslynItem))
+            {
+                change.Properties!.TryGetValue("LSPSnippet", out var lspSnippetText);
```

Comment:
1. consider making the properties *always* non null.  you can always set it to an immutable empty dictionary.
2. use a constant for "LSPSnippet" so the setter/getter agree on this value.,

Reply:
Does this mean we generate snippet text for every snippet completion item pre-emptively? If so, is there a req on doing it pre-emptively? I half would have assumed we'd only ever generate the snippet text until the last second when we're about to commit it.

Reply:
good point